### PR TITLE
App integrations

### PR DESCRIPTION
**Pull request by royrutto**

### CODE REVIEW

File: src/milia/api/io.cljs [hunk 1]
```diff
@@ -46,10 +46,13 @@
 (defn token->headers
   "Builds request headers for the HTTP request by adding
   Authorization, X-CSRFToken and Cache-control headers where necessary"
-  [& {:keys [get-crsftoken? must-revalidate? accept-header]}]
+  [& {:keys [get-crsftoken? must-revalidate? accept-header auth-token]}]
```
Comment (pld) [inline src/milia/api/io.cljs]: this is getting destructured as `auth-token`, but being set as `token`, https://github.com/onaio/milia/pull/146/files#diff-73181e15b22f8475f61619687e90b50dR68

**Comment by pld** [comment]
the `io` and `http` changes need tests

File: src/milia/api/io.clj [hunk 2]
```diff
@@ -39,12 +39,15 @@
 (defn- req+auth
   "Add authorization to options"
   [req]
-  (let [{:keys [temp-token token username password]} *credentials*]
+  (let [{:keys [temp-token token username password]} *credentials*
+        {:keys [auth-token]} req]
```
Comment (pld) [inline src/milia/api/io.clj]: pre-existing but we should call `req` `http-options` here and in the `build-request` function, that's what it really is

File: src/milia/api/io.clj [hunk 3]
```diff
@@ -39,12 +39,15 @@
...
     (if (or temp-token token)
```
Comment (pld) [inline src/milia/api/io.clj]: should we be in the first branch if `auth-token` is set?

File: src/milia/api/io.clj [hunk 4]
```diff
@@ -39,12 +39,15 @@
...
       (assoc req
-             :headers {"Authorization" (if temp-token
-                                         (str "TempToken " temp-token)
-                                         (str "Token " token))})
+             :headers {"Authorization" (if auth-token
```
Comment (pld) [inline src/milia/api/io.clj]: the conditional can perhaps be more simply written as:

```clojure
(join " " (if (and temp-token (not auth-token))
            ["TempToken" token]
            ["Token" (or auth-token token)]))
```

given you import `clojure.string :refer [join]`

**Comment by royrutto** [comment]
@pld this is ready for CR again.

**Pull request merged**